src/dbus/client.rs

```diff
@@ -15,19 +18,90 @@ impl Client {
         Ok(Client { conn })
     }
 
-    /// start_unit starts a specific unit under systemd. See https://www.freedesktop.org/wiki/Software/systemd/dbus
+    /// start_transient_unit is a higher level API for starting a unit
+    /// for a specific container under systemd.
+    /// See https://www.freedesktop.org/wiki/Software/systemd/dbus
     /// for more details.
-    pub fn start_unit(&self, unit_name: &str, _properties: Vec<&str>) -> Result<()> {
+    pub fn start_transient_unit_for_container(
+        &self,
+        container_name: &str,
+        unit_name: &str,
+        linux_resources: LinuxResources,
+    ) -> Result<()> {
         let proxy = self.conn.with_proxy(
-            "org.freedesktop.systemd1.Manager",
-            "/",
+            "org.freedesktop.systemd1",
+            "/org/freedesktop/systemd1",
             Duration::from_millis(5000),
         );
-        let (_job_id,): (i32,) = proxy.method_call(
-            "org.freedesktop.systemd1.Manager",
-            "StartTransientUnit",
-            (unit_name, "replace"),
-        )?;
+
+        let mut properties = Vec::<Property>::new();
+        properties.push(
+            PropDescription::new(format!("youki container {}", container_name).as_str()).into(),
+        );
+
+        let slice = "machine.slice";
+        // if we create a slice, the parent is defined via a Wants=
+        // otherwise, we use Slice=
+        if unit_name.ends_with("slice") {
+            properties.push(PropWants::new(slice).into());
+        } else {
+            properties.push(PropSlice::new(slice).into());
+        }
+
+        proxy.start_transient_unit(unit_name, "replace", properties, vec![])?;
         Ok(())
     }
 }
+
+type Property = (&'static str, Variant<Box<dyn RefArg>>);
```

Comment:
I'm considering using a procedural macro to derive all these helper structs and their impls.

Reply:
Or I can keep it simple and just write the tuples directly.